### PR TITLE
Enhance/support prompt for caption generation

### PR DESCRIPTION
**Pull request by HYLcool**
As the title says.

### CODE REVIEW

**Review by yxdyc — APPROVED** [review]
LGTM

File: data_juicer/ops/mapper/generate_caption_mapper.py [hunk 1]
```diff
@@ -157,9 +164,19 @@ def _process_single_sample(self, ori_sample):
             # generated_text_candidates_single_chunk[i][j] indicates
             # the $i$-th generated candidate for the $j$-th image
 
+            # construct prompts
+            if self.prompt_key \
+                    and isinstance(ori_sample[self.prompt_key], str):
+                # check prompt_key is not None, and it's a str in the sample
+                prompt_texts = [ori_sample[self.prompt_key]] * len(image_chunk)
+            elif self.prompt and isinstance(self.prompt, str):
```
Comment (yxdyc) [inline data_juicer/ops/mapper/generate_caption_mapper.py:172]: maybe the line 172 and 168 need to be re-ordered? users may want to quickly change the prompt by overriding the prompt-keys, if both of them are specified. BTW, we can warn users with logger for this case

Comment (BeachWang) [reply data_juicer/ops/mapper/generate_caption_mapper.py:172]: maybe we need to check if the ori_sample contains this key.

Comment (HYLcool) [reply data_juicer/ops/mapper/generate_caption_mapper.py:172]: > maybe we need to check if the ori_sample contains this key.

Actually, in line 175, if self.prompt_key doesn't exist in ori_sample, ori_sample[self.prompt_key] will return None, and it's not an instance of str, so the condition will be False.

Comment (HYLcool) [reply data_juicer/ops/mapper/generate_caption_mapper.py:179]: > maybe the line 172 and 168 need to be re-ordered? users may want to quickly change the prompt by overriding the prompt-keys, if both of them are specified. BTW, we can warn users with logger for this case

A warning will be reported when both `prompt` and `prompt_key` are set in the initialization method.

**Pull request merged**